src/test/java/org/stellar/sdk/Sep10ChallengeTest.java

```diff
@@ -46,7 +50,1296 @@ public void testChallenge() throws IOException {
+
+    long now = System.currentTimeMillis() / 1000L;
+    long end = now + 300;
+    TimeBounds timeBounds = new TimeBounds(now, end);
+
+    byte[] nonce = new byte[48];
+    SecureRandom random = new SecureRandom();
+    random.nextBytes(nonce);
+    BaseEncoding base64Encoding = BaseEncoding.base64();
+    byte[] encodedNonce = base64Encoding.encode(nonce).getBytes();
+
+    Account sourceAccount = new Account(server.getAccountId(), -1L);
+    ManageDataOperation manageDataOperation1 = new ManageDataOperation.Builder(anchorName + " auth", encodedNonce)
+            .build();
+
+    Operation[] operations = new Operation[]{manageDataOperation1};
+    Transaction transaction = new Transaction(
+            sourceAccount.getAccountId(),
+            100 * operations.length,
+            sourceAccount.getIncrementedSequenceNumber(),
+            operations,
+            Memo.none(),
+            timeBounds,
+            network
+    );
+    transaction.sign(server);
+    String challenge = transaction.toEnvelopeXdrBase64();
+
+    try {
+      Sep10Challenge.readChallengeTransaction(challenge, server.getAccountId(), Network.TESTNET);
+      fail();
+    } catch (InvalidSep10ChallengeException e) {
+      assertEquals("Operation should have a source account.", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testReadChallengeTransactionInvalidDataValueWrongEncodedLength() throws IOException {
+    KeyPair server = KeyPair.random();
+    KeyPair client = KeyPair.random();
+    String anchorName = "Stellar Test";
+    Network network = Network.TESTNET;
+
+    long now = System.currentTimeMillis() / 1000L;
+    long end = now + 300;
+    TimeBounds timeBounds = new TimeBounds(now, end);
+
+    byte[] nonce = new byte[32];
+    SecureRandom random = new SecureRandom();
+    random.nextBytes(nonce);
+    BaseEncoding base64Encoding = BaseEncoding.base64();
+    byte[] encodedNonce = base64Encoding.encode(nonce).getBytes();
+
+    Account sourceAccount = new Account(server.getAccountId(), -1L);
+    ManageDataOperation manageDataOperation1 = new ManageDataOperation.Builder(anchorName + " auth", encodedNonce)
+            .setSourceAccount(client.getAccountId())
+            .build();
+
+    Operation[] operations = new Operation[]{manageDataOperation1};
+    Transaction transaction = new Transaction(
+            sourceAccount.getAccountId(),
+            100 * operations.length,
+            sourceAccount.getIncrementedSequenceNumber(),
+            operations,
+            Memo.none(),
+            timeBounds,
+            network
+    );
+    transaction.sign(server);
+    String challenge = transaction.toEnvelopeXdrBase64();
+
+    try {
+      Sep10Challenge.readChallengeTransaction(challenge, server.getAccountId(), Network.TESTNET);
+      fail();
+    } catch (InvalidSep10ChallengeException e) {
+      assertEquals("Random nonce encoded as base64 should be 64 bytes long.", e.getMessage());
+    }
+  }
+
```

Comment:
❗️ Could we add a test here that tests what happens if the data value base64 that is in the challenge transaction is corrupted?

This is the `TestReadChallengeTx_invalidDataValueCorruptBase64` test in the Go SDK:
https://github.com/stellar/go/blob/f4e38b5/txnbuild/transaction_test.go#L1512-L1539

Reply:
https://github.com/stellar/java-stellar-sdk/pull/264#discussion_r381269520 and https://github.com/stellar/java-stellar-sdk/blob/dbd69a425b0ac43ce2f95c1e8d2948f2bc095449/src/test/java/org/stellar/sdk/Sep10ChallengeTest.java#L543L579

Reply:
Looks good, thanks!